src/tools/recvfuzz/recvfuzz.cpp

```diff
@@ -505,25 +683,222 @@ void buildInitialPacket(CXPLAT_SOCKET* Binding, CXPLAT_ROUTE Route, int64_t* Pac
     }
 }
 
-void fuzzHandshakePacket() {
-// TODO
-}
-
 void fuzz(CXPLAT_SOCKET* Binding, CXPLAT_ROUTE Route) {
     int64_t PacketCount = 0;
     int64_t TotalByteCount = 0;
-    uint8_t mode;
+    uint8_t mode = 1;
     uint64_t StartTimeMs = CxPlatTimeMs64();
+    uint64_t SrcCid;
+    PacketParams HandshakePacketParams = {
+        sizeof(uint64_t),
+        sizeof(uint64_t),
+        nullptr,
+        nullptr,
+        0,
+        1
+    };
+    CxPlatEventInitialize(&(RecvPacketEvent), TRUE, FALSE);
+    HandshakePacketParams.FrameTypes[0] = QUIC_FRAME_CRYPTO;
+    TlsContext HandshakeClientContext;
+    HandshakeClientContext.CreateContext();
+    bool ServerHello = FALSE;
+    uint8_t recvBuffer[8192];
+    uint32_t bufferoffset = 0;
+    bool handshakeComplete = FALSE;
+    bool indicateMode = TRUE;
     while (CxPlatTimeDiff64(StartTimeMs, CxPlatTimeMs64()) < RunTimeMs) {
-        mode = 0; //(uint8_t)GetRandom(2);
         if (mode == 0) {
-            buildInitialPacket(Binding, Route, &PacketCount, &TotalByteCount);
+            PacketParams InitialPacketParams = {
+                sizeof(uint64_t),
+                sizeof(uint64_t),
+                nullptr,
+                nullptr,
+                0,
+                1
+            };
+            if (indicateMode == TRUE) {
+                printf("Sending Initial Packets\n");
+                indicateMode = FALSE;
+            }
+            InitialPacketParams.PacketType = QUIC_INITIAL_V1;
+            InitialPacketParams.FrameTypes[0] = QUIC_FRAME_CRYPTO;
+            InitialPacketParams.mode = 0;
+            sendPacket(Binding, Route, &PacketCount, &TotalByteCount, &InitialPacketParams, true);
         } else if (mode == 1) {
-            fuzzHandshakePacket();
+            if (!handshakeComplete) {
+                if (!ServerHello) {
+                    do {
+                        CxPlatRandom(sizeof(uint64_t), &SrcCid);
+                        HandshakePacketParams.SourceCid = (uint8_t *)&SrcCid;
+                        HandshakeClientContext.CreateContext(SrcCid);
+                        HandshakeClientContext.ProcessData();
+                        HandshakePacketParams.PacketType = QUIC_INITIAL_V1;
+                        HandshakePacketParams.mode = 1;
+                        printf("Sending Client Hello\n");
+                        sendPacket(Binding, Route, &PacketCount, &TotalByteCount, &HandshakePacketParams, false, &HandshakeClientContext);
+                    } while (!RecvPacketEvent.WaitTimeout(200) && CxPlatTimeDiff64(StartTimeMs, CxPlatTimeMs64()) < RunTimeMs);
+                    RecvPacketEvent.Reset();
```

Comment:
What happens is it takes longer than 200 ms for the packet to come back? Does it mess things up?

Reply:
What I mean is, what if you continue, but the previous packet _then_ comes in and the event gets set?

Reply:
Even if it does, that new packet wont be processed cause we updated the source cid and that needs to match for us to process the packet

Reply:
There is a check for that later